tools/accuracy_checker/accuracy_checker/adapters/centernet.py

```diff
@@ -0,0 +1,161 @@
+
+        topk_inds = topk_inds % (height * width)
+        topk_ys = (topk_inds / width).astype(np.int32).astype(np.float)
+        topk_xs = (topk_inds % width).astype(np.int32).astype(np.float)
+
+        topk_scores = topk_scores.reshape((batch, -1))
+        topk_ind = np.argpartition(topk_scores, -K, axis=1)[:, -K:]
+        topk_score = topk_scores[np.arange(topk_scores.shape[0])[:, None], topk_ind]
+        topk_clses = (topk_ind / K).astype(np.int32)
+        topk_inds = CTDETAdapter._gather_feat(
+            topk_inds.reshape((batch, -1, 1)), topk_ind).reshape((batch, K))
+        topk_ys = CTDETAdapter._gather_feat(topk_ys.reshape((batch, -1, 1)), topk_ind).reshape((batch, K))
+        topk_xs = CTDETAdapter._gather_feat(topk_xs.reshape((batch, -1, 1)), topk_ind).reshape((batch, K))
+
+        return topk_score, topk_inds, topk_clses, topk_ys, topk_xs
+
+    @staticmethod
+    def _nms(heat, kernel=3):
+        def max_pool2d(A, kernel_size, padding=1, stride=1):
+            A = np.pad(A, padding, mode='constant')
+            output_shape = ((A.shape[0] - kernel_size)//stride + 1,
+                            (A.shape[1] - kernel_size)//stride + 1)
+            kernel_size = (kernel_size, kernel_size)
+            A_w = as_strided(A, shape=output_shape + kernel_size,
+                                strides=(stride*A.strides[0],
+                                        stride*A.strides[1]) + A.strides)
+            A_w = A_w.reshape(-1, *kernel_size)
+
+            return A_w.max(axis=(1, 2)).reshape(output_shape)
+
+        pad = (kernel - 1) // 2
+
+        hmax = np.array([max_pool2d(channel, kernel, pad) for channel in heat[0]])
+        hmax = np.expand_dims(hmax, axis=0)
+        keep = (hmax == heat)
+        return heat * keep
+
+    @staticmethod
+    def _transform_preds(coords, center, scale, output_size):
+        def affine_transform(pt, t):
+            new_pt = np.array([pt[0], pt[1], 1.], dtype=np.float32).T
+            new_pt = np.dot(t, new_pt)
+            return new_pt[:2]
+
+        target_coords = np.zeros(coords.shape)
+        trans = CenterNetAffineTransformation.get_affine_transform(center, scale, 0, output_size, inv=1)
+        for p in range(coords.shape[0]):
+            target_coords[p, 0:2] = affine_transform(coords[p, 0:2], trans)
+        return target_coords
+
+    @staticmethod
+    def _transform(dets, center, scale, heigth, width):
+        for i in range(dets.shape[0]):
+            dets[i, :, :2] = CTDETAdapter._transform_preds(
+                dets[i, :, 0:2], center[i], scale[i], (width, heigth))
+            dets[i, :, 2:4] = CTDETAdapter._transform_preds(
+                dets[i, :, 2:4], center[i], scale[i], (width, heigth))
+        return dets
+
+    def process(self, raw, identifiers=None, frame_meta=None):
+        result = []
+        hm = self._extract_predictions(raw, frame_meta)[self.hm]
```

Comment:
could you please extract predictions only one time and after that use already extracted?